medusa/databases/main_db.py

```diff
@@ -840,3 +840,47 @@ def shift_history_qualities(self):
                 'UPDATE history SET quality = ? WHERE quality = ?;',
                 [new_quality, quality]
             )
+
+
+class AddEpisodeWatchedField(ShiftQualities):
+    """Add episode watched field."""
+
+    def test(self):
+        """Test if the version is at least 44.12"""
+        return self.connection.version >= (44, 12)
+
+    def execute(self):
+        utils.backup_database(self.connection.path, self.connection.version)
+
+        log.info(u'Adding new watched field in the tv_episodes table')
+        self.connection.action('DROP TABLE IF EXISTS tmp_tv_episodes;')
+        self.connection.action('ALTER TABLE tv_episodes RENAME TO tmp_tv_episodes;')
+
+        self.connection.action(
+            'CREATE TABLE IF NOT EXISTS tv_episodes '
+            '(episode_id INTEGER PRIMARY KEY, showid NUMERIC, indexerid INTEGER, indexer INTEGER, '
+            'name TEXT, season NUMERIC, episode NUMERIC, description TEXT, airdate NUMERIC, hasnfo NUMERIC, '
+            'hastbn NUMERIC, status NUMERIC, quality NUMERIC, location TEXT, file_size NUMERIC, release_name TEXT, '
+            'subtitles TEXT, subtitles_searchcount NUMERIC, subtitles_lastsearch TIMESTAMP, '
+            'is_proper NUMERIC, scene_season NUMERIC, scene_episode NUMERIC, absolute_number NUMERIC, '
+            'scene_absolute_number NUMERIC, version NUMERIC DEFAULT -1, release_group TEXT, '
+            'manually_searched NUMERIC, watched NUMERIC);'
+        )
+
+        # Re-insert old values, setting the new quality column to the invalid value of -1
+        self.connection.action(
+            'INSERT INTO tv_episodes '
+            '(showid, indexerid, indexer, name, season, episode, description, airdate, hasnfo, '
+            'hastbn, status, quality, location, file_size, release_name, subtitles, subtitles_searchcount, '
+            'subtitles_lastsearch, is_proper, scene_season, scene_episode, absolute_number, scene_absolute_number, '
+            'version, release_group, manually_searched, watched) '
+            'SELECT showid, indexerid, indexer, '
+            'name, season, episode, description, airdate, hasnfo, '
+            'hastbn, status, -1 AS quality, location, file_size, release_name, '
```

Comment:
I don't understand why you are changing the quality fields to `-1`?

Reply:
Thats a foobar